src/groupeddataframe/splitapplycombine.jl

```diff
@@ -257,25 +262,56 @@ Last Group (1 row): a = 4
 │     │ Int64 │ Int64 │
 ├─────┼───────┼───────┤
 │ 1   │ 4     │ 12    │
+
+julia> map(nrow, gd)
+GroupedDataFrame with 4 groups based on key: a
+First Group (1 row): a = 1
+│ Row │ a     │ nrow  │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 2     │
+⋮
+Last Group (1 row): a = 4
+│ Row │ a     │ nrow  │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 4     │ 2     │
 ```
 
 See [`by`](@ref) for more examples.
 """
 function Base.map(f::Union{Base.Callable, Pair}, gd::GroupedDataFrame)
     if length(gd) > 0
-        idx, valscat = _combine(f, gd, nothing)
-        keys = _names(gd.parent)[gd.cols]
+        # here we know that parent(gd) has at least 1 column
+        if f isa Pair || f === nrow
+            if f isa Pair && first(f) isa Tuple
+                Base.depwarn("passing a Tuple $(first(f)) as column selector is deprecated" *
+                             ", use a vector $(collect(first(f))) instead", :combine)
+                source_cols, (fun, out_col) = normalize_selection(index(parent(gd)),
+                                                                  collect(first(f)) => last(f))
+            else
+                source_cols, (fun, out_col) = normalize_selection(index(parent(gd)), f)
+            end
+            if isagg(source_cols => fun)
+                idx, valscat = _combine([source_cols => fun], gd, [out_col])
+            else
+                idx, valscat = _combine(source_cols => last(f), gd, nothing)
```

Comment:
Is that correct?
```suggestion
                idx, valscat = _combine(source_cols => fun, gd, nothing)
```

Reply:
What I have written is correct.

Your proposal would make `combine(:a => fun => :b, gd)` "forget" the `:b` part (as we can check how to handle it only at run-time so we have to pass a whole `last(f)` to `_combine`).